command_line/powder_calibrate.py

```diff
@@ -0,0 +1,467 @@
+            self.geometry.poni2 = beam_px[0] * self.detector.pixel2
+        print(self.geometry)
+
+    def set_calibrant(self, name):
+        self.calibrant_name = name
+        self.calibrant = pfCalibrant(name)
+        self.calibrant.wavelength = _convert_units(self.data.wavelength, "A", "m")
+        print("\n Using:\n----- \n", self.calibrant)
+
+    def calibrate_with_calibrant(
+        self,
+        name,
+        num_rings=4,
+        fix=["rot1", "rot2", "rot3", "wavelength"],
+        ref=False,
+        verbose=False,
+    ):
+        self.set_calibrant(name)
+        if verbose:
+            self.show_calibrant(self.geometry)
+
+        sg = pfSingleGeometry(
+            label=name + " calibrant",
+            image=self.data.image,
+            calibrant=self.calibrant,
+            geometry=self.geometry,
+        )
+
+        sg.extract_cp(max_rings=num_rings)
+        if verbose:
+            show_fit(sg)
+
+        if ref:
+            sg.geometry_refinement.refine2(fix=fix)
+            if verbose:
+                show_fit(sg)
+
+            ai = sg.get_ai()
+            int2 = ai.integrate2d_ng(
+                self.data.image, 500, 360, unit="q_nm^-1", filename="integrated.edf"
+            )
+            if verbose:
+                pfjupyter.plot2d(int2, calibrant=self.calibrant)
+                plt.show()
+
+    def tricky_fit_with_calibrant(
+        self,
+        name,
+        num_rings=4,
+        fix=["rot1", "rot2", "rot3", "wavelength"],
+        verbose=False,
+    ):
+        self.calibrate_with_calibrant(name=name, num_rings=4, fix=fix, verbose=verbose)
+        beam = self.beam_from_fit_circle()
+        self.update_geometry(beam_px=beam)
+        self.calibrate_with_calibrant(
+            name=name, num_rings=4, fix=fix, ref=True, verbose=verbose
+        )
+
+    def show_calibrant(self, geometry):
+        fig, ax = subplots(1, 2, figsize=(8, 5))
+        pfjupyter.display(self.data.image, label="Experiment image", ax=ax[0])
+
+        # for smaller wavelengths reduce the blurring of the rings
+        if self.data.wavelength < 1e-1:
+            w = 1e-7
+        else:
+            w = 1e-3
+
+        cal_img = self.calibrant.fake_calibration_image(geometry, W=w)
+        pfjupyter.display(cal_img, label=self.calibrant_name + " calibrant", ax=ax[1])
+        plt.show()
+
+    def show_fit(self, geometry):
+        pfjupyter.display(sg=geometry)
+        plt.show()
+
+    def save_refined_geometry(self, file_name):
+        if self.sg:
+            self.sg.geometry_refinement.save(file_name)
+        else:
+            pass
+
+
+if __name__ == "__main__":
+
+    Al_data = DialsData(
+        expt_file="/home/elena/Work/Diamond/2021/Al_data/Yun_cameralength_cal/cameralength_cal/Al-standard_0p67_5s/imported.expt",
+        refl_file="/home/elena/Work/Diamond/2021/Al_data/Yun_cameralength_cal/cameralength_cal/Al-standard_0p67_5s/strong.refl",
+    )
```

Comment:
When I ran `dials.powder_calibrate` with no arguments I got
```
FileNotFoundError: [Errno 2] No such file or directory: '/home/elena/Work/Diamond/2021/Al_data/Yun_cameralength_cal/cameralength_cal/Al-standard_0p67_5s/imported.expt'
```
Would prefer for new programs to use [the boilerplate](https://github.com/dials/dials/wiki/Writing-a-DIALS-program) to get consistent presentation of help messages, option switches etc.